respy/likelihood.py

```diff
@@ -283,21 +284,65 @@ def log_like_obs(state_space, df, base_draws_est, optim_paras, options):
     # over all periods.
     prob_type = np.multiply.reduceat(prob_obs, idx_indiv_first_obs)
 
+    # Calculate the probabilities for all types based on an individual's initial
+    # characteristics.
+    type_probabilities = (
+        predict_multinomial_logit(optim_paras["type_prob"], type_covariates)
+        if n_types > 1
+        else 1
+    )
+
     # Multiply each individual-type contribution with its type-specific shares and sum
     # over types to get the likelihood contribution for each individual.
-    contribs = (prob_type * type_shares).sum(axis=1)
+    contribs = (prob_type * type_probabilities).sum(axis=1)
 
     contribs = np.clip(np.log(contribs), -HUGE_FLOAT, HUGE_FLOAT)
 
     return contribs
 
 
 def _convert_choice_variables_from_categorical_to_codes(df, options):
-    # Recode choices to model codes. It is not possible to use ``.cat.codes`` because
-    # the codes might be in a different order than for the model required which is
-    # choices_w_exp_w_wag, choices_w_exp_wo_wage, choices_wo_exp_wo_wage.
+    """Recode choices to choice codes in the model.
+
+    We cannot use ``.cat.codes`` because order might be different. The model requires an
+    order of ``choices_w_exp_w_wag``, ``choices_w_exp_wo_wage``,
+    ``choices_wo_exp_wo_wage``.
+
+    See also
+    --------
+    respy.pre_processing.model_processing._order_choices
+
+    """
     choices_to_codes = {choice: i for i, choice in enumerate(options["choices"])}
-    df.Choice = df.Choice.replace(choices_to_codes).astype(int)
-    df.Lagged_Choice = df.Lagged_Choice.replace(choices_to_codes).astype(int)
+    df.choice = df.choice.replace(choices_to_codes).astype(int)
```

Comment:
`np.uint16` or even `np.uint8` should be enough. Is there a reason to use normal ints?

Reply:
No, there is not. Did not think about it.